addon/PlacesProvider.js

```diff
@@ -420,21 +402,42 @@ Links.prototype = {
     let links = yield this.executePlacesQuery(sqlQuery, {
       columns: [
         "bookmarkGuid",
-        "favicon",
         "frecency",
         "guid",
         "lastVisitDate",
-        "mimeType",
         "title",
         "type",
         "url"
       ],
       params: {limit}
     });
 
+    links = yield this._getFavicons(links);
     return this._processLinks(links);
   }),
 
+  _getFavicons: Task.async(function*(links) {
+    if (links.length) {
+      let faviconPromises = [];
+      links.forEach(item => {
+        faviconPromises.push(new Promise(resolve => {
+          PlacesUtils.favicons.getFaviconDataForPage(NetUtil.newURI(item.url), (iconuri, len, data, mime) => {
```

Comment:
We probably don't need to add a test for the bad uri case, but at least making sure we're actually getting expected favicon data now using `getFaviconDataForPage`